lib/project/pro_motion/pm_application.rb

```diff
@@ -118,9 +118,42 @@ def alert(options={}, &block)
     # Send user to native Texting
     # app.sms("555-555-5555")
     def sms(phone_number)
-      sms_intent = Android::Content::Intent.new("android.intent.action.VIEW")
-      sms_intent.setData(Android::Net::Uri.fromParts("sms", phone_number.to_s, nil))
-      find.activity.startActivity(sms_intent)
+      mp "[BP Deprecated] use app.launch(sms: #{phone_number}) over app.sms"
+      launch(sms: phone_number)
+    end
+
+    # Launch native services via intent
+    # app.launch(sms: '5045558008')
+    # app.launch(web: 'http://giphy.com')
+    # app.launch(email: 'your@mom.com')
+    # app.launch(email: 'your@mom.com', subject: "Hey Chica", message: "Howdy")
+    # app.launch(chooser: 'I hope you have a nice day!')
+    def launch(command={})
+      action_view = "android.intent.action.VIEW"
+      action_send = "android.intent.action.SEND"
+      launch_intent = case command.keys.first #TODO: fragile, re-evaluate
+      when :sms
+        sms_intent = Android::Content::Intent.new(action_view)
+        sms_intent.setData(Android::Net::Uri.fromParts("sms", command[:sms].to_s, nil))
+      when :email
+        email_intent = Android::Content::Intent.new(action_view)
+        email_string = "mailto:#{command[:email]}"
+        email_string += "?subject=#{command[:subject].to_s}"
+        email_string += "&body=#{command[:message].to_s}"
+        email_intent.setData(Android::Net::Uri.parse(email_string))
+      when :web
+        web_intent = Android::Content::Intent.new(action_view)
+        web_intent.setData(Android::Net::Uri.parse(command[:web]))
+      when :chooser
+        message_intent = Android::Content::Intent.new(action_send)
+        message_intent.type = "text/plain"
+        message_intent.putExtra("android.intent.extra.TEXT", command[:chooser].to_s) if command[:chooser]
+        Android::Content::Intent.createChooser(message_intent, nil)
+      else
+        mp "[BP Warning] Unsupported launch type '#{command.keys.first}'"
```

Comment:
Havent tested, but `mp` might have a non-nil return value which would cause the `launch_intent` to get triggered.  Perhaps just putting a `nil` after this line would fix?